distributed/protocol/compression.py

```diff
@@ -153,32 +155,31 @@ def maybe_compress(
     """
     if compression == "auto":
         compression = default_compression
-
     if not compression:
         return None, payload
-    if len(payload) < min_size:
-        return None, payload
-    if len(payload) > 2**31:  # Too large, compression libraries often fail
+
+    # Store size as it is used in a few cases.
+    payload_nbytes = nbytes(payload)
+
+    # Either too small to bother or too large, compression libraries often fail
+    if not (min_size < payload_nbytes < 2**31):
```

Comment:
I know you didn't decide this, but not compressing because too big seems pretty odd.

Reply:
Yeah this traces back to issue ( https://github.com/dask/distributed/issues/366 ), which observed this behavior and PR ( https://github.com/dask/distributed/pull/367 ) that fixed it.

We've seen similar issues in Zarr in particular with Blosc and LZ4. Please see PR ( https://github.com/zarr-developers/numcodecs/pull/81 ).

Blosc isn't relevant for us any more here, but LZ4 is.

If I had to guess, these libraries are using `int32` to store some buffer size, which has maximum value of `2**32 - 1` (similar to what is checked here). Now IDK if this is a simple oversight of using `int` instead of `size_t` or if there are more significant issues algorithmically that would need to be tackled in these compressors for larger buffers.

Independently we may choose to solve this problem differently like by breaking up small buffers and compressing them in pieces.

In any event this is probably out-of-scope for this PR, but could be interesting to see how we can improve on this.

Reply:
I don't expect we'll hit it either way; but I owuld have thought splitting is the way to go, or choosing a compressor that explicitly has framing.

Reply:
It can come up, but agree we could do better at handling this case.